src/metpy/calc/thermo.py

```diff
@@ -1632,6 +1632,45 @@ def saturation_equivalent_potential_temperature(pressure, temperature):
     return units.Quantity(th_es, units.kelvin)
 
 
+@exporter.export
+@preprocess_and_wrap(
+    wrap_like='temperature',
+    broadcast=('pressure', 'temperature', 'dewpoint')
+)
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def wetbulb_potential_temperature(pressure, temperature, dewpoint):
+    r"""Calculate wet-bulb potential temperature.
+
+    This calculation must be given an air parcel's pressure, temperature, and dewpoint.
+    The implementation uses the formula outlined in [DaviesJones2008]_:
+
+    First, theta-e is calculated
+    then use the formula from [DaviesJones2008]_
```

Comment:
Let's include the formula here:
```suggestion
    This calculation must be given an air parcel's pressure, temperature, and dewpoint.
    The implementation uses the formula outlined in [DaviesJones2008]_.  First, theta-e is
    calculated, then used in a rational polygon fit:

    .. math:: \theta_w = \theta_e - exp(\frac{a_0 + a_1 x + a_2 x^2 + a_3 x^3 + a_4 x^4}{1 + b_1 x + b_2 x^2 + b_3 x^3 + b_4 x^4}})

    where :math:`x = \theta_e / 273.15 K`. When :math:`\theta_e <= -173.15 k`, :math:`\theta_w = \theta_e`.
```